src/bpffeature.h

```diff
@@ -165,9 +165,8 @@ class BPFfeature {
       int* outfd = nullptr);
   bool try_load_btf(const void* btf_data, size_t btf_size);
 
-  BTF btf_ = BTF({ "vmlinux" });
-
   BPFnofeature no_feature_;
+  BTF* btf_;
```

Comment:
Having this be a bare pointer creates ambiguity about ownership, since the `BPFfeature` type can be instantiated independently. Maybe just make the BTF type a `shared_ptr` across the board?

Otherwise, you might want to a way to bind the lifetime here. (If none of those are nice suggestions, than I'd say at minimum you'd want to document the semantics on the constructor and here.)

Reply:
Yeah, a shared ptr is a good idea - let me look into that.